microsoft-edge/webview2/how-to/winrt-from-js.md

```diff
@@ -0,0 +1,336 @@
+---
+title: Call native-side WinRT code from web-side code
+description: How to pass WinRT host objects to JavaScript using the AddHostObjectToScript API for WebView2 apps.
+author: MSEdgeTeam
+ms.author: msedgedevrel
+ms.topic: conceptual
+ms.prod: microsoft-edge
+ms.technology: webview
+ms.date: 05/02/2022
+---
+# Call native-side WinRT code from web-side code
+
+Your web-side JavaScript code can access the methods and properties of your WinRT host app, by using the _WebView2 WinRT JS Projection tool_.  (For short, this is sometimes called the _WV2WinRT_ tool.)  For more information about why you'd want to do this, see the introduction of [Call native-side code from web-side code](hostobject.md).
+
+This article walks you through the following main steps:
+
+*  Create a **WinRTAdapter** project especially to contain the _WebView2 WinRT JS Projection tool_.
+
+*  Run the WebView2 WinRT JS Projection tool to generate C++/WinRT source code for selected namespaces or classes.
+
+*  In the WinUI 2 (UWP) sample WebView2 browser app project (**webview2_sample_uwp**), call `AddHostObjectToScript`. 
+
+*  Your web-side JavaScript code, and the DevTools console, then calls methods and properties on the host object.
+
+
+<!-- ====================================================================== -->
+## WinRT vs. .NET
+
+This article is for the WinRT WebView2 APIs, not for the .NET WebView2 APIs.  The C# code in this article will build, but not run, for the .NET WebView2 APIs.  If you were to call `AddHostObjectToScript` using this article's C# code for .NET WebView2 APIs, that would produce an error message.
+
+The WebView2 WinRT JS Projection tool is needed when projecting WinRT objects, because WinRT by default doesn't support `iDispatch`, which WebView2's Win32 and .NET platforms support.
+
+For .NET, see [Call native-side code from web-side code](hostobject.md) instead of this article.
+
+
+<!-- ====================================================================== -->
+## Let's get started!
+
+First let's find a native object (class) that we're interested in calling from JavaScript code.  For this example, we'll use the WinRT `Language` class, which is in the `Windows.Globalization` namespace, for Windows UWP applications.  The [Language Class](/uwp/api/windows.globalization.language) enables getting information from the client's native OS.
+
+In the WebView2 host app, the web-side JavaScript code can then access properties and methods on the `Language` object that's in the native-side code.  At the end of this example walkthrough, you'll use the **Console** of Microsoft Edge DevTools to test reading the host's `displayName` property of the `Language` class:
+
+```javascript
+(new chrome.webview.hostObjects.sync.Windows.Globalization.Language("en-US")).displayName
```

Comment:
Accessing the host object can be done as setup for the script, which is also what we expect devs to do. Actual calls to projected objects would look more like:
```javascript
// Somewhere early in setup:
const Windows = chrome.webview.hostObjects.sync.Windows;

(new Windows.Globalization.Language("en-US")).displayName;
```